### PR TITLE
[Feature] CUE Based Workflow Step

### PR DESCRIPTION
**Pull request by hongchaodeng**
This is to implement the CUE based workflow steps based on [this design](https://github.com/oam-dev/kubevela/blob/master/design/vela-core/workflow_policy.md)

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Pulling up this PR for triggering the tests only.

File: pkg/controller/core.oam.dev/v1alpha2/application/revision.go [hunk 1]
```diff
@@ -48,40 +47,25 @@ import (
...
 const (
-	// ConfigMapKeyResources is the key in ConfigMap Data field for containing data of resources
-	ConfigMapKeyResources = "resources"
+	// ConfigMapKeyComponents is the key in ConfigMap Data field for containing data of components
+	ConfigMapKeyComponents = "components"
+	// ConfigMapKeyPolicy is the key in ConfigMap Data field for containing data of policies
+	ConfigMapKeyPolicy = "policies"
 )
 
 func (h *AppHandler) createResourcesConfigMap(ctx context.Context,
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/revision.go:56]: will this refactor afftect the `AppDeployment` logic? @hongchaodeng

File: pkg/controller/core.oam.dev/v1alpha2/application/revision.go [hunk 2]
```diff
@@ -105,6 +89,21 @@ func (h *AppHandler) createResourcesConfigMap(ctx context.Context,
 	return h.r.Client.Create(ctx, cm)
 }
 
+// SprintComponentManifest formats and returns the resulting string.
+func SprintComponentManifest(cm *types.ComponentManifest) string {
+	cl := map[string]interface{}{
+		"StandardWorkload": string(util.MustJSONMarshal(cm.StandardWorkload)),
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/revision.go:95]: please use a const value for `StandardWorkload`.

File: pkg/controller/core.oam.dev/v1alpha2/application/revision.go [hunk 3]
```diff
@@ -105,6 +89,21 @@ func (h *AppHandler) createResourcesConfigMap(ctx context.Context,
...
+	}
+
+	trs := []string{}
+	for _, tr := range cm.Traits {
+		trs = append(trs, string(util.MustJSONMarshal(tr)))
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/revision.go:100]: we should write tests to cover these kinds of logic.

**Review by wonderflow — CHANGES_REQUESTED** [review]
please improve the test coverage, especially for some main logic(exclude error return)

File: pkg/controller/core.oam.dev/v1alpha2/application/revision.go (hunk 1 above)
Comment (hongchaodeng) [reply pkg/controller/core.oam.dev/v1alpha2/application/revision.go:56]: No.

File: pkg/controller/core.oam.dev/v1alpha2/application/revision.go (hunk 3 above)
Comment (hongchaodeng) [reply pkg/controller/core.oam.dev/v1alpha2/application/revision.go:100]: This has been covered

**Review by wonderflow — APPROVED** [review]

**Comment by hongchaodeng** [comment]
LGTM

**Pull request merged**